src/patternfly/components/Tabs/tabs.scss

```diff
@@ -112,167 +126,305 @@
   max-width: 100%;
   overflow-x: auto;
   -webkit-overflow-scrolling: touch;
-
-  // Right border on last tab
-  .pf-c-tabs__item:last-of-type .pf-c-tabs__button::before {
-    border-right-width: var(--pf-c-tabs__item--BorderWidth);
-  }
 }
 
-// Tab items
+// Tabs item
 .pf-c-tabs__item {
+  display: flex;
   flex: none;
+  align-items: stretch;
+
+  // Current
+  &.pf-m-current {
+    .pf-c-tabs__button,
+    .pf-c-tabs__button:hover::after {
+      --pf-c-tabs__button--Color: var(--pf-c-tabs__item--m-current--c-tabs__button--Color);
+      --pf-c-tabs__button--after--BorderColor: var(--pf-c-tabs__item--m-current--c-tabs__button--after--BorderColor);
+    }
+  }
 
-  // Tab item hover state
-  &:not(.pf-m-current):hover .pf-c-tabs__button::after {
+  // If focussed and hovered
+  &:not(.pf-m-current) .pf-c-tabs__button:focus::after {
+    --pf-c-tabs__button--after--BorderColor: transparent;
+  }
+}
+
+
+// Buttons
+.pf-c-tabs__button,
+.pf-c-tabs__scroll-button {
+  border: 0;
+
+  // Button borders
+  &::before,
+  &::after {
     position: absolute;
     top: 0;
     right: 0;
     bottom: 0;
     left: 0;
-    margin-left: var(--pf-c-tabs__item--BorderWidth);
     content: "";
-    border-top: var(--pf-c-tabs__item--m-current--BorderTopWidth) solid var(--pf-c-tabs__item--hover--Color);
-  }
-
-  // Tab item current state
-  &.pf-m-current .pf-c-tabs__button {
-    color: var(--pf-c-tabs__item--m-current--Color);
-
-    &::before {
-      border-bottom-color: var(--pf-c-tabs__item--BackgroundColor);
-    }
-
-    &::after {
-      position: absolute;
-      top: 0;
-      right: 0;
-      bottom: 0;
-      left: 0;
-      margin-left: var(--pf-c-tabs__item--BorderWidth);
-      content: "";
-      border-top: var(--pf-c-tabs__item--m-current--BorderTopWidth) solid var(--pf-c-tabs__item--m-current--Color);
-    }
-  }
-
-  // Secondary tab states
-  .pf-m-tabs-secondary & .pf-c-tabs__button {
-    // Remove current state and hover state
-    &::before,
-    &::after,
-    &:hover::after {
-      content: none;
-    }
-  }
-
-  // Secondary tab hover state
-  .pf-m-tabs-secondary &:not(.pf-m-current) .pf-c-tabs__button:hover {
-    color: var(--pf-c-tabs__item--hover--Color);
+    border-style: var(--pf-c-tabs__button--before--BorderStyle);
+    border-width: 0;
   }
 }
 
-
 // Tab button
 .pf-c-tabs__button {
   position: relative;
-  display: inline-block;
+  display: flex;
+  flex: 1;
   padding: var(--pf-c-tabs__button--PaddingTop) var(--pf-c-tabs__button--PaddingRight) var(--pf-c-tabs__button--PaddingBottom) var(--pf-c-tabs__button--PaddingLeft);
-  font-weight: var(--pf-c-tabs__button--FontWeight);
   color: var(--pf-c-tabs__button--Color);
   user-select: none;
-  background: var(--pf-c-tabs__button--Background);
+  background-color: var(--pf-c-tabs__button--Background);
```

Comment:
Can the property name match the var? Probably most flexible to update the var to `--pf-c-tabs__button--BackgroundColor` in case we just want to alter one individual background property with a modifier later on without having to override all the properties using `background` shorthand.